src/ql/com/tests/topology_test.cc

```diff
@@ -0,0 +1,156 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest/doctest.h"
+
+#include "ql/com/topology.h"
+
+namespace ql {
+namespace com {
+
+class TopologyTest {};
+
+TEST_CASE_FIXTURE(TopologyTest, "Single core grid 2x3") {
+    std::uint64_t qubit_count = 6;
+
+/*
+
+        0 --- 1 --- 2
+        |     |     |
+        5 --- 4 --- 3
+
+*/
+
+    // This API is weird: I don't understand why "edges" need to be specified, when "form" is "xy".
```

Comment:
@jvansomeren do you reckon that specifying a grid structure should implicitely declare the grid qubit connectivity, not just coordinates?